src/items/associated-items.md

```diff
@@ -205,22 +205,47 @@ types cannot be defined in [inherent implementations] nor can they be given a
 default implementation in traits.
 
 An *associated type declaration* declares a signature for associated type
-definitions. It is written as `type`, then an [identifier], and
-finally an optional list of trait bounds.
+definitions. It is written in one of the following forms, where `Assoc` is the
+name of the associated type, `Params` is a comma-separated list of type,
+lifetime or const parameters, `Bounds` is a plus-separated list of trait bounds
+on the associated type, and `WhereBounds` is a comma-separated list of bounds on
+parameters:
+
+```rust,ignore
+type Assoc;
+type Assoc: Bounds;
+type Assoc<Params>;
+type Assoc<Params>: Bounds;
+type Assoc<Params> where WhereBounds;
+type Assoc<Params>: Bounds where WhereBounds;
+```
 
 The identifier is the name of the declared type alias. The optional trait bounds
 must be fulfilled by the implementations of the type alias.
 There is an implicit [`Sized`] bound on associated types that can be relaxed using the special `?Sized` bound.
 
-An *associated type definition* defines a type alias on another type. It is
-written as `type`, then an [identifier], then an `=`, and finally a [type].
+An *associated type definition* defines a type alias on for the implementation
+of a trait on a type. They are written similarly to an *associated type declaration*,
+but cannot contain `Bounds`, but instead must contain a `Type`:
+
+```rust,ignore
+type Assoc = Type;
+type Assoc<Params> = Type<Params>;
```

Comment:
```suggestion
type Assoc<Params> = Type;
```

This is inconsistent with the other examples, no?

Reply:
Yeah, I specifically wanted to highlight that you can use those `Params` in the defined type. But maybe it's just too confusing

Reply:
Yeah, I can see that. I could also see people interpreting it to mean that the type *has* to have that form (e.g., `Vec<T>` or something).

Reply:
Could do 

```rust
type Assoc<Params> = Type; // the type `Type` here may reference `Params`
```

if you want to make it extra clear.